flink-libraries/flink-table/src/main/scala/org/apache/flink/table/api/queryConfig.scala

```diff
@@ -92,6 +100,27 @@ class StreamQueryConfig private[table] extends QueryConfig {
     this
   }
 
+  /**
+    * Specifies a deferred computation time for deferred computation, i.e., fires the
```

Comment:
Specifies an offset for the point in time when the first result of a time-based computation is computed. For example, a tumbling window of one hour that ends at 13:00 would usually compute its first result at 13:00. With a firstResultTimeOffset of 15 minutes, the first result would be computed at 13:15.

A positive firstResultTimeOffset parameter can be used to include late arriving records into the result of an event-time based computation. Negative offset values are not supported yet.

Later, a negative offset will allow to compute early results, i.e., an offset of -45 minutes would compute the first and early result of the hourly tumbling window that ends at 13:00 at 12:15.